src/transformers/models/data2vec/modeling_data2vec_vision.py

```diff
@@ -1180,7 +1180,9 @@ def forward(
 
         # only keep certain features, and reshape
         # note that we do +1 as the encoder_hidden_states also includes the initial embeddings
-        features = [feature for idx, feature in enumerate(encoder_hidden_states) if idx + 1 in self.config.out_indices]
+        features = [
+            feature for idx, feature in enumerate(encoder_hidden_states) if idx + 1 in self.config.semantic_out_indices
+        ]
```

Comment:
We can't make this breaking change - it completely changes the behaviour if someone did the following: 

```
model = Data2VecVisionForSemanticSegmentation("facebook/data2vec-vision-base-ft1k", output_indices=(-2, -1))
```